Mathlib/Condensed/Light/TopCatAdjunction.lean

```diff
@@ -0,0 +1,129 @@
+all the maps from light profinite sets `S` to `X(*)`, corresponding to elements of `X(S)`.
+In other words, the topology coinduced by the map `LightCondSet.coinducingCoprod` above. -/
+local instance : TopologicalSpace (X.val.obj ⟨LightProfinite.of PUnit⟩) :=
+  TopologicalSpace.coinduced (coinducingCoprod X) inferInstance
+
+/-- The object part of the functor `LightCondSet ⥤ TopCat`  -/
+def LightCondSet.toTopCat : TopCat.{u} := TopCat.of (X.val.obj ⟨LightProfinite.of PUnit⟩)
+
+namespace LightCondSet
+
+variable {X} {Y : LightCondSet} (f : X ⟶ Y)
+
+/-- The map part of the functor `LightCondSet ⥤ TopCat`  -/
+@[simps]
+def toTopCatMap : X.toTopCat ⟶ Y.toTopCat where
+  toFun := f.val.app ⟨LightProfinite.of PUnit⟩
+  continuous_toFun := by
+    rw [continuous_coinduced_dom]
+    apply continuous_sigma
+    intro ⟨S, x⟩
+    simp only [Function.comp_apply, coinducingCoprod]
+    have : (fun (a : S) ↦ f.val.app ⟨LightProfinite.of PUnit⟩ (X.val.map (S.const a).op x)) =
+        (fun (a : S) ↦ Y.val.map (S.const a).op (f.val.app ⟨S⟩ x)) :=
+      funext fun a ↦ NatTrans.naturality_apply f.val (S.const a).op x
+    rw [this]
+    suffices ∀ (i : (T : LightProfinite.{u}) × Y.val.obj ⟨T⟩),
+        Continuous (fun (a : i.fst) ↦ Y.coinducingCoprod ⟨i, a⟩) from this ⟨_, _⟩
+    rw [← continuous_sigma_iff]
+    apply continuous_coinduced_rng
+
+end LightCondSet
+
+/-- The functor `LightCondSet ⥤ TopCat`  -/
+@[simps]
+def lightCondSetToTopCat : LightCondSet.{u} ⥤ TopCat.{u} where
+  obj X := X.toTopCat
+  map f := toTopCatMap f
+
+namespace LightCondSet
+
+/-- The counit of the adjunction `lightCondSetToTopCat ⊣ topCatToLightCondSet` -/
+def topCatAdjunctionCounit (X : TopCat.{u}) : X.toLightCondSet.toTopCat ⟶ X where
+  toFun x := x.1 PUnit.unit
+  continuous_toFun := by
+    rw [continuous_coinduced_dom]
+    continuity
+
+/-- The counit of the adjunction `lightCondSetToTopCat ⊣ topCatToLightCondSet` is always bijective,
+but not an isomorphism in general (the inverse isn't continuous unless `X` is sequential).
+-/
+def topCatAdjunctionCounitEquiv (X : TopCat.{u}) : X.toLightCondSet.toTopCat ≃ X where
+  toFun := topCatAdjunctionCounit X
+  invFun x := ContinuousMap.const _ x
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+lemma topCatAdjunctionCounit_bijective (X : TopCat.{u}) :
+    Function.Bijective (topCatAdjunctionCounit X) :=
+  (topCatAdjunctionCounitEquiv X).bijective
+
+/-- The unit of the adjunction `lightCondSetToTopCat ⊣ topCatToLightCondSet` -/
+@[simps val_app val_app_apply]
+def topCatAdjunctionUnit (X : LightCondSet.{u}) : X ⟶ X.toTopCat.toLightCondSet where
+  val := {
+    app := fun S x ↦ {
+      toFun := fun s ↦ X.val.map (S.unop.const s).op x
+      continuous_toFun := by
+        suffices ∀ (i : (T : LightProfinite.{u}) × X.val.obj ⟨T⟩),
+          Continuous (fun (a : i.fst) ↦ X.coinducingCoprod ⟨i, a⟩) from this ⟨_, _⟩
+        rw [← continuous_sigma_iff]
+        apply continuous_coinduced_rng }
```

Comment:
I think it would be better to extract the `app` field as a separate definition in this case, and the naturality as a lemma. This way we could have better control over the `simps` lemmas that are generated.

Reply:
The `val_app` simp lemma is crucial to prove naturality of the unit. It's pretty ugly, of the form 
```lean
  X.topCatAdjunctionUnit.val.app S x =
    { toFun := fun s ↦ X.val.map (S.unop.const s).op x, continuous_toFun := ⋯ }
```
I tried extracting the `app` field as a separate definition, but then this lemma wouldn't fire, unless I stated it explicitly

Reply:
Should I try to make it work without this lemma?

Reply:
Oh, in that case let's just leave it as is. It would probably work if you tagged the `app` definition with `simp` (not `simps`), but that's not ideal.